read_data.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+)
+
+type Establishment struct {
+	ID           string  `json:"id"`
+	TYPE         string  `json:"type"`
+	NAME         string  `json:"name"`
+	BUILDING     string  `json:"building"`
+	ROOM         string  `json:"room"`
+	URL          string  `json:"url"`
+	X_COORDINATE float64 `json:"x"`
+	Y_COORDINATE float64 `json:"y"`
+}
+
+var db *gorm.DB
+
+type App struct {
+	DB *gorm.DB
+}
+
+func (a *App) Initialize(dbDriver string, dbURI string) {
+	db, err := gorm.Open(dbDriver, dbURI)
+	if err != nil {
+		panic("failed to connect database")
+	}
+	a.DB = db
+
+	db.AutoMigrate(&Establishment{})
+}
+
+func (a *App) listEstHandler(c *gin.Context) {
+	//db := c.MustGet("db").(*gorm.DB)
+	var spots []Establishment
+
+	if result := a.DB.Find(&spots); result.Error != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{
+			"error": result.Error.Error(),
+		})
+		return
+	}
+
+	//a.DB.Find(&restaurants)
+
+	c.JSON(http.StatusOK, &spots)
+}
+
+func (a *App) createEstablishments(c *gin.Context) {
+	var res []Establishment
+
+	if err := c.ShouldBindJSON(&res); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": err.Error(),
+		})
+		return
+	}
+
+	if result := a.DB.Create(&res); result.Error != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{
+			"error": result.Error.Error(),
+		})
+		return
+	}
+
+	c.JSON(http.StatusCreated, &res)
+}
+
+func main() {
+
+	var err error
+	db, err = gorm.Open("sqlite3", "seproj.db")
+
+	if err != nil {
+		panic("failed to connect database")
+	}
+
+	a := &App{}
+	a.Initialize("sqlite3", "seproj.db")
+
+	r := gin.New()
+	r.GET("/", func(c *gin.Context) {
+		c.JSON(http.StatusOK, gin.H{
+			"message": "Hello World!",
+		})
+	})
+
+	//api := r.Group("/api/restaurants")
+	//{
+	//api.GET("/", a.listEstHandler)
+	r.POST("/restaurants", a.createEstablishments)
+	r.GET("/restaurants", a.listEstHandler)
+
+	//}
```

Comment:
Everything looks good, but can you uncomment this and use the `r.Group`? We will have other endpoints for users and reviews and it might be helpful to have them organized

Also we are changing the name to establishments, like so

```
	api := r.Group("/api/establishments")
	{
	api.POST("/", a.createEstablishments)
	api.GET("/", a.listEstHandler)

	}
```

That way to make the request, you take the first part of the group `/api/establishments` and then what ever you passed to the api.POST method `/`, so we can make calls with `GET "/api/establishments/"`

Reply:
Sure! I have made the changes and used the r.Group 
We can make calls with GET "/establishments/"